src/cpu/aarch64/acl_inner_product.hpp

```diff
@@ -257,10 +268,17 @@ struct acl_inner_product_fwd_t : public primitive_t {
                     return status::unimplemented;
             }
 
+            const memory_desc_t weights_md_received = weights_md_;
             acl_utils::reorder_to_weight_format(aip.wei_tensor_info,
                     weights_md_, expected_weight_format, inner_dim, o_dim,
                     remaining_dims, {});
 
+            if ((weights_format_kind_received == format_kind::blocked)
+                    && !(dnnl_memory_desc_equal(
+                            &weights_md_received, &weights_md_))) {
+                return status::unimplemented;
```

Comment:
Please could you use the `ACL_CHECK_SUPPORT` macro, along with some helpful words, maybe something like:
`specific blocked format not supported by ACL, use format_kind_t::any to find a supported blocked format for your platform`